File: go/longest_consecutive_sequence/longest_consecutive_sequence.go
package longestconsecutivesequence

import "fmt"

// Given an unsorted array of integers nums, return the length of the longest consecutive elements sequence.

// You must write an algorithm that runs in O(n) time.

func longestConsecutive(nums []int) int {
	var m = make(map[int]int)
	var max int
	for _, num := range nums {
		if _, exists := m[num-1]; exists {
			m[num] = m[num-1] + 1
		} else {
			m[num] = 1
		}
		fmt.Println(m)
	}
	for _, v := range m {
		if v > max {
			max = v
		}
	}
	return max
}

File: old/python/longest_consecutive_sequence/longest_consecutive_sequence.py
from typing import List


class LongestConsecutive:
    def longest_consecutive(self, nums: List[int]) -> int:
        ans, nums_set = 0, set(nums)

        for num in nums:
            if num not in nums_set:
                continue
            nums_set.remove(num)
            current_ans = 1
            i = 1

            while num + i in nums_set:
                nums_set.remove(num + i)
                current_ans += 1
                i += 1

            while num - i in nums_set:
                nums_set.remove(num - i)
                current_ans += 1
                i += 1
            ans = max(ans, current_ans)
        return ans
